x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/ChangePasswordRequestBuilder.java

```diff
@@ -41,22 +41,23 @@ public ChangePasswordRequestBuilder username(String username) {
         return this;
     }
 
-    public static char[] validateAndHashPassword(SecureString password) {
+    public static char[] validateAndHashPassword(SecureString password, String hashingAlgorithm) {
+        final Hasher hasher = Hasher.resolve(hashingAlgorithm, Hasher.BCRYPT);
```

Comment:
I suggest we use the `Hasher` enum in place of the `String` value of the setting. Meaning, resolve to enum asap. In this case, resolve enum in `RestChangePasswordAction`'s ctr .
Also, I don't think resolve requires a default value, as there is a default for the setting value.

Reply:
I wonder if there should be a `public static <T extends Enum<T>> Setting<T> Setting#enumSetting(String key, Class<T> class, Function<Settings, T> defaultValue)`